packages/playwright-test/src/matchers/toMatchSnapshot.ts

```diff
@@ -214,7 +214,10 @@ class SnapshotHelper<T extends ImageComparatorOptions> {
 
     if (expected !== undefined) {
       writeFileSync(this.expectedPath, expected);
-      this.testInfo.attachments.push({ name: addSuffixToFilePath(this.snapshotName, '-expected'), contentType: this.mimeType, path: this.expectedPath });
+      // Note: use actual snapshot path in the snapshots directory instead of the one
+      // copied over to the `test-results/` folder.
+      // This way we can use this path later on when we need to update snapshot expectation.
+      this.testInfo.attachments.push({ name: addSuffixToFilePath(this.snapshotName, '-expected'), contentType: this.mimeType, path: this.snapshotPath });
       output.push(`Expected: ${colors.yellow(this.expectedPath)}`);
```

Comment:
- Perhaps change output to `snapshotPath` as well?
- More broadly, I'd remove `this.expectedPath` and `snapshotPath` everywhere, only calculating it here with a comment "additionally copy the expectation to the old location for backwards compatibility, so that one can upload test-results directory and have all the data inside"